### DISCUSSION TITLE
Custom modifier for ObservableProperty

### DISCUSSION THREAD

**Discussion by Kiryuumaru** [CommunityToolkit/dotnet#349] [General]
Hi, is there a feature to specify the ObservableProperty modifier using the source generator?
````
[ObservableObject]
public class Person
{
    [ObservableProperty(SetterModifier.Private)]
    private string? name;
}
````
To generate
````
public class Person
{
    public string? Name
    {
        get => name;
        private set => SetProperty(ref name, value);
    }
}
````

**Comment by Sergio0694** [comment]
No, this is not currently supported. We plan to support that in a future release once we get partial properties in C# 🙂

**Reply by Kiryuumaru** [reply]
Yayy, glad to know you plan to support this in a future release 💖

**Reply by rexhiebert** [reply]
Are we there yet? Or is there a workaround for private setters?

**Reply by echoix** [reply]
There was a comment yesterday on the blocking issue explaining a bit where it is at : https://github.com/dotnet/csharplang/issues/6420#issuecomment-1745771890

**Reply by Balkoth** [reply]
This is definitely a feature i would like to see.

**Reply by ArieGato** [reply]
partial properies have been implemented in c# 13. [Reference](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties#properties-with-backing-fields). Will the feature be implemented now?

**Reply by kebox7** [reply]
This is already implemented in the version [8.4.0](https://github.com/CommunityToolkit/dotnet/releases/tag/v8.4.0)

![image](https://github.com/user-attachments/assets/a998cdef-bc56-4085-a617-0da8d73e77b8)

**Reply by Insire** [reply]
This requires 
```xml
<PropertyGroup>
    <LangVersion>preview</LangVersion>
</PropertyGroup>
```
to be set tho.

With .NET 10 this will be remedied
